4.x Test logging converter for OpenTelemetry-related testing does not handle span attributes with an empty string value

## Environment Details
* Helidon Version: 4.x
* Helidon SE or Helidon MP
* JDK version:
* OS:

----------

## Problem Description

(found by @barchetta !)

Support for OpenTelemetry tracing includes a Helidon test logging converter which translates span data into JSON written to an OTel class's logger.

That test converter does not properly tolerate the case where a span attribute had an empty string value.

```list
[ERROR] Errors: 
[ERROR]   MainTest>AbstractMainTest.testSpanForNamedGreeting:70 » IllegalArgument Cannot convert attribute with key 'helidon.socket' because the inner
key datatype is not recognized: []
```
## Steps to reproduce
Build the `examples/telemetry/config` example. The unit test in the example app fails with the exception above.

--- a/telemetry/testing/tracing/pom.xml
+++ b/telemetry/testing/tracing/pom.xml
@@ -52,10 +52,15 @@
             <artifactId>hamcrest-all</artifactId>
         </dependency>
         <dependency>
-        <groupId>org.junit.jupiter</groupId>
-        <artifactId>junit-jupiter-api</artifactId>
-        <scope>test</scope>
-    </dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.eclipse.parsson</groupId>
+            <artifactId>parsson</artifactId>
+            <scope>test</scope>
+        </dependency>
 
     </dependencies>
 

--- a/telemetry/testing/tracing/src/main/java/io/helidon/telemetry/testing/tracing/JsonLogConverterImpl.java
+++ b/telemetry/testing/tracing/src/main/java/io/helidon/telemetry/testing/tracing/JsonLogConverterImpl.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2025 Oracle and/or its affiliates.
+ * Copyright (c) 2025, 2026 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -43,11 +43,11 @@ class JsonLogConverterImpl implements JsonLogConverter {
     /**
      * The format emits all attribute values as strings, but each attribute value has a key indicating its datatype.
      */
-    private static final Map<String, Function<String, ?>> DATA_TYPE_MAPPERS = Map.of("intValue", Integer::parseInt,
+    private static final Map<String, Function<String, Object>> DATA_TYPE_MAPPERS = Map.of("intValue", Integer::parseInt,
                                                                                      "longValue", Long::parseLong,
                                                                                      "floatValue", Float::parseFloat,
                                                                                      "doubleValue", Double::parseDouble,
-                                                                                     "stringValue", Function.identity());
+                                                                                     "stringValue", String::toString);
 
     private final Logger logger;
     private final TestLogHandler testLogHandler;
@@ -57,7 +57,11 @@ class JsonLogConverterImpl implements JsonLogConverter {
         // The OTLP JSON span exporter places JSON blocks for each span in the logger.
         // Attach our handler to that logger so we can check the exported data easily.
 
-        logger = Logger.getLogger(OtlpJsonLoggingSpanExporter.class.getName());
+        this(Logger.getLogger(OtlpJsonLoggingSpanExporter.class.getName()));
+    }
+
+    JsonLogConverterImpl(Logger testLogger) {
+        logger = testLogger;
         testLogHandler = new TestLogHandler();
         logger.addHandler(testLogHandler);
     }
@@ -228,9 +232,12 @@ class JsonLogConverterImpl implements JsonLogConverter {
         JsonObject value = jsonAttributeElement.get("value").asJsonObject();
 
         /*
-        Every attribute value, regardless of type, is represented in the JSON as a string in double quotes.
+        Normally, every attribute value, regardless of type, is represented in the JSON as a string in double quotes.
         Below, we use {@code value.getString(k)} to strip off the double quotes so the various datatype parsers can work on
         the enclosed string.
+
+        An attribute with an empty value (an empty string) will not have any value keyset in the JSON which OpenTelemetry
+        generates, so for those return a map entry with an empty string value.
          */
         return value.keySet()
                 .stream()
@@ -238,13 +245,7 @@ class JsonLogConverterImpl implements JsonLogConverter {
                                                         DATA_TYPE_MAPPERS.get(k)
                                                                 .apply(value.getString(k))))
                 .findFirst()
-                .orElseThrow(() ->
-                                     new IllegalArgumentException(String.format(
-                                             """
-                                                     Cannot convert attribute with key '%s' because the inner
-                                                     key datatype is not recognized: %s""",
-                                             key,
-                                             value.keySet())));
+                .orElseGet(() -> new AbstractMap.SimpleEntry<>(key, ""));
 
     }
 

--- a/telemetry/testing/tracing/src/test/java/io/helidon/telemetry/testing/tracing/JsonLogConverterTest.java
+++ b/telemetry/testing/tracing/src/test/java/io/helidon/telemetry/testing/tracing/JsonLogConverterTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2025 Oracle and/or its affiliates.
+ * Copyright (c) 2025, 2026 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -21,6 +21,8 @@ import java.util.logging.Logger;
 import org.junit.jupiter.api.Test;
 
 import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasEntry;
 import static org.hamcrest.Matchers.hasSize;
 
 class JsonLogConverterTest {
@@ -39,5 +41,133 @@ class JsonLogConverterTest {
                 .forEach(span -> logger.log(Level.INFO, "Second msg"));
         assertThat("Fetch of spans during update", testLogHandler.resourceSpans(), hasSize(2));
 
+    }
+
+    @Test
+    void testEmptyStringValueAttribute() throws Exception {
+        /*
+        Mimic the default (empty) helidon socket attribute.
+        {"key":"helidon.socket","value":{}}
+         */
+
+        Logger logger = Logger.getLogger(JsonLogConverterTest.class.getName());
+        try (var converter = new JsonLogConverterImpl(logger)) {
+
+            logger.log(Level.INFO, """
+                    {
+                       "resource" : {
+                         "attributes" : [ {
+                           "key" : "service.name",
+                           "value" : {
+                             "stringValue" : "otel-config-example"
+                           }
+                         }, {
+                           "key" : "telemetry.sdk.language",
+                           "value" : {
+                             "stringValue" : "java"
+                           }
+                         }, {
+                           "key" : "telemetry.sdk.name",
+                           "value" : {
+                             "stringValue" : "opentelemetry"
+                           }
+                         }, {
+                           "key" : "telemetry.sdk.version",
+                           "value" : {
+                             "stringValue" : "1.58.0"
+                           }
+                         }, {
+                           "key" : "x",
+                           "value" : {
+                             "stringValue" : "x-value"
+                           }
+                         }, {
+                           "key" : "y",
+                           "value" : {
+                             "intValue" : "9"
+                           }
+                         } ]
+                       },
+                       "scopeSpans" : [ {
+                         "scope" : {
+                           "name" : "otel-config-example",
+                           "attributes" : [ ]
+                         },
+                         "spans" : [ {
+                           "traceId" : "6edae103f4de519e341ed47e84888e6f",
+                           "spanId" : "89076986e7a0b27e",
+                           "name" : "GET",
+                           "kind" : 2,
+                           "startTimeUnixNano" : "1769634903759517000",
+                           "endTimeUnixNano" : "1769634931221218583",
+                           "attributes" : [ {
+                             "key" : "server.address",
+                             "value" : {
+                               "stringValue" : "helidon-unit"
+                             }
+                           }, {
+                             "key" : "net.host.name",
+                             "value" : {
+                               "stringValue" : "helidon-unit"
+                             }
+                           }, {
+                             "key" : "user_agent.original",
+                             "value" : {
+                               "stringValue" : "Helidon 4.4.0-SNAPSHOT"
+                             }
+                           }, {
+                             "key" : "url.path",
+                             "value" : {
+                               "stringValue" : "/greeting/Joe"
+                             }
+                           }, {
+                             "key" : "http.request.method",
+                             "value" : {
+                               "stringValue" : "GET"
+                             }
+                           }, {
+                             "key" : "helidon.socket",
+                             "value" : { }
+                           }, {
+                             "key" : "server.port",
+                             "value" : {
+                               "stringValue" : "8080"
+                             }
+                           }, {
+                             "key" : "http.response.status_code",
+                             "value" : {
+                               "stringValue" : "404"
+                             }
+                           }, {
+                             "key" : "url.scheme",
+                             "value" : {
+                               "stringValue" : "http"
+                             }
+                           }, {
+                             "key" : "net.host.port",
+                             "value" : {
+                               "intValue" : "8080"
+                             }
+                           } ],
+                           "events" : [ ],
+                           "links" : [ ],
+                           "status" : { },
+                           "flags" : 257
+                         } ]
+                       } ]
+                     }""");
+
+            var logSpan = converter.resourceSpans(1).getFirst()
+                    .scopeSpans().getFirst()
+                    .logSpans().getFirst();
+
+            /*
+            The presence of helidon.socket with an empty value caused a problem in the converter that should be fixed.
+             */
+            assertThat("Attributes", logSpan.attributes(), allOf(
+                    hasEntry("server.address", "helidon-unit"),
+                    hasEntry("url.path", "/greeting/Joe"),
+                    hasEntry("helidon.socket", "")));
+        }
     }
 }

